homeassistant/components/light/group.py

```diff
@@ -0,0 +1,280 @@
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed for a group light."""
+        return False
+
+    async def _async_send_message(self, service, **kwargs):
+        """Send a message to all entities in the group."""
+        for entity_id in self._entity_ids:
+            payload = deepcopy(kwargs)
+            payload[ATTR_ENTITY_ID] = entity_id
+            await self.hass.services.async_call(light.DOMAIN, service, payload)
+
+    async def async_turn_on(self, **kwargs):
+        """Forward the turn_on command to all lights in the group."""
+        await self._async_send_message(SERVICE_TURN_ON, **kwargs)
+
+    async def async_turn_off(self, **kwargs):
+        """Forward the turn_off command to all lights in the group."""
+        await self._async_send_message(SERVICE_TURN_OFF, **kwargs)
+
+    async def async_update(self):
+        """Query all members and determine the group state."""
+        states = self._child_states()
+        on_states = [state for state in states if state.state == STATE_ON]
+
+        self._state = _determine_on_off_state(states)
+
+        self._brightness = _reduce_attribute(on_states, ATTR_BRIGHTNESS)
+
+        self._xy_color = _reduce_attribute(
+            on_states, ATTR_XY_COLOR, reduce=_mean_tuple)
+
+        self._rgb_color = _reduce_attribute(
+            on_states, ATTR_RGB_COLOR, reduce=_mean_tuple)
+        if self._rgb_color is not None:
+            self._rgb_color = tuple(map(int, self._rgb_color))
+
+        self._white_value = _reduce_attribute(on_states, ATTR_WHITE_VALUE)
+
+        self._color_temp = _reduce_attribute(on_states, ATTR_COLOR_TEMP)
+        self._min_mireds = _reduce_attribute(
+            states, ATTR_MIN_MIREDS, default=154, reduce=min)
+        self._max_mireds = _reduce_attribute(
+            states, ATTR_MAX_MIREDS, default=500, reduce=max)
+
+        self._effect_list = None
+        all_effect_lists = list(
+            _find_state_attributes(states, ATTR_EFFECT_LIST))
+        if all_effect_lists:
+            # Merge all effects from all effect_lists with a union merge.
+            self._effect_list = list(set().union(*all_effect_lists))
+
+        self._effect = None
+        all_effects = list(_find_state_attributes(on_states, ATTR_EFFECT))
+        if all_effects:
+            # Report the most common effect.
+            effects_count = Counter(itertools.chain(all_effects))
+            self._effect = effects_count.most_common(1)[0][0]
+
+        self._supported_features = 0
+        for support in _find_state_attributes(states, ATTR_SUPPORTED_FEATURES):
+            # Merge supported features by emulating support for every feature
+            # we find.
+            self._supported_features |= support
+        # Bitwise-and the supported features with the GroupedLight's features
+        # so that we don't break in the future when a new feature is added.
+        self._supported_features &= SUPPORT_GROUP_LIGHT
```

Comment:
`SUPPORT_GROUP_LIGHT` is a confusing name since `SUPPORT_` constants are used to describe the features of an entity, while here it is describing what support this entity has, if one of its children has that support.

I think that there is no need for `SUPPORT_GROUP_LIGHT` and this line at all ?

Reply:
> I think that there is no need for SUPPORT_GROUP_LIGHT and this line at all ?

I mean the reason was that if some entity decided to add support for some weird supported effect, that wouldn't be reported to the front-end through our `reduce_attribute` code, we shouldn't just fail silently. But I'm open to changing this back.

Reply:
You are right. Let's be conservative and only support the features we know to support.